### DISCUSSION TITLE
Revision-related push preventing required check in Pull Request

### DISCUSSION THREAD

**Discussion by jethas-bennettjones** [community/community#26895] [Actions]
<p>Hi all,</p>
<p>I am currently struggling with a workflow that I would like some help with.</p>
<p>My current workflow operates like this:<br>
First the user pushes the code to GitHub, an Action script with pushes to branches other than <code>main</code> gets triggered and as part of its process, it versions the front- and back-end projects with pre-release versions. The workflow continues with linting, compiling, testing, etc., then when it’s all successful, it does a <code>push</code> so that the pre-release versions are on the branch.<br>
This all works fine until I want to put down successful passes as requirements for PRs. What I end up with is a successful pass followed immediately by a version change push that doesn’t trigger any Actions. The PR only looks at the latest run and because it neither passed nor failed, the PR is blocked from being merged.</p>
<p>Here’s an example of what my PR’s commit history looks like:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/717be2fad261db9874de0ab898ec02dded35eba4.png" data-download-href="/uploads/short-url/gbVnyXttZmRh2x0AwPIgVe73YbO.png?dl=1" title="Capture" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/62298251/181100494-e0f3f8f4-9dea-4593-9912-7249f9929bb7.png" alt="Capture" data-base62-sha1="gbVnyXttZmRh2x0AwPIgVe73YbO" width="690" height="117" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/717be2fad261db9874de0ab898ec02dded35eba4_2_690x117.png, https://aws1.discourse-cdn.com/github/original/2X/7/717be2fad261db9874de0ab898ec02dded35eba4.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/7/717be2fad261db9874de0ab898ec02dded35eba4.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/717be2fad261db9874de0ab898ec02dded35eba4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture</span><span class="informations">848×145 10.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’d appreciate any help or ideas on how I can make this workflow work as it should.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jethas-bennettjones">@jethas-bennettjones</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<aside class="quote no-group" data-username="jethas-bennettjones" data-post="1" data-topic="131278" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 jethas-bennettjones:</div>
<blockquote>
<p>What I end up with is a successful pass followed immediately by a version change push that doesn’t trigger any Actions.</p>
</blockquote>
</aside>
<p>Could you please let me know how you do the version change? And if it doesn’t trigger any workflow, it should not be in the pull request check run list.</p>
<p>Is it possible to add the version change step into workflow and it will be treated as a workflow?</p>
<p>It’s recommened to share your repo or a sample repo here for further investigation.</p>
<p>Thanks</p>

**Comment by jethas-bennettjones** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a>,</p>
<p>The repo is private so I can’t share a link to it directly, but here is a sample workflow file:</p>
<pre><code class="lang-auto">name: Build (without versioning)
env:
  webapp_path: Project.WebApp

on:
  push:
    branches-ignore:
      - main

jobs:
  build-angular:
    runs-on: ubuntu-latest
    outputs:
      version: ${{steps.version.outputs.version}}
    steps:
      - uses: actions/checkout@v2

      - name: Setup NPM
        uses: actions/setup-node@v1
        with:
          node-version: "12"
          registry-url: "https://npm.pkg.github.com"

      - name: Get npm cache directory
        id: npm-cache
        run: |
          echo "::set-output name=dir::$(npm config get cache)"

      - name: Cache npm cache
        uses: actions/cache@v2
        with:
          path: ${{ steps.npm-cache.outputs.dir }}
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-node-

      - name: Install npm packages
        working-directory: ./${{env.webapp_path}}/ClientApp
        run: npm ci
        env:
          NODE_AUTH_TOKEN: ${{secrets.PULL_NPM}}
          CYPRESS_INSTALL_BINARY: 0

      - name: Lint Angular app
        working-directory: ./${{env.webapp_path}}/ClientApp
        run: npm run lint

      - name: Bump Angular app version
        id: pre-version
        run: |
          echo "::set-output name=version::$(npm version prerelease)"
        working-directory: ./${{env.webapp_path}}/ClientApp

      - id: version
        run: |
          version=$(echo ${{steps.pre-version.outputs.version}} | cut -dv -f2)
          echo "::set-output name=version::$version"

      - name: Compile Angular app
        working-directory: ./${{env.webapp_path}}/ClientApp
        run: npm run build:ci --node-flags --max-old-space-size=7168 -- -c=test

      - name: Test Angular app
        uses: cypress-io/github-action@v2
        with:
          working-directory: ./${{env.webapp_path}}/ClientApp
          start: npx angular-http-server -g -p 4200 --path dist
          wait-on: "http://localhost:4200"

      - name: Publish Cypress Videos
        uses: actions/upload-artifact@v2
        if: always()
        with:
          name: Cypress Videos
          path: ${{env.webapp_path}}/ClientApp/cypress/videos/

      - name: Publish app
        uses: actions/upload-artifact@v2
        with:
          name: angular-app
          path: ./${{env.webapp_path}}/ClientApp/dist

  test-dotnet:
    runs-on: windows-latest
    env:
      DOTNET_NOLOGO: true
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-dotnet@v1
        with:
          dotnet-version: "3.1.x" # SDK Version to use; x will use the latest version of the 3.1 channel
          source-url: https://nuget.pkg.github.com/bennettjones/index.json
        env:
          NUGET_AUTH_TOKEN: ${{secrets.PULL_NPM}}

      - name: Setup NuGet
        uses: NuGet/setup-nuget@v1.0.2

      - name: Cache NUGET packages
        uses: actions/cache@v2
        with:
          path: ~/.nuget/packages
          key: ${{ runner.os }}-nuget-${{ hashFiles('**/${{env.webapp_path}}.Web.csproj')}}

      - name: Create node_modules folder # Needed so that the build process skips looking for "npm"
        working-directory: ./${{env.webapp_path}}/ClientApp
        run: New-Item -Type Directory -Name node_modules

      - name: Test App (Unit Tests)
        run: dotnet test --filter "FullyqualifiedName!~IntegrationTests"


      # - name: Test App (Integration)
      #   run: dotnet test --filter "FullyQualifiedName~IntegrationTests"

  package:
    needs: [build-angular, test-dotnet]
    runs-on: windows-latest
    env:
      DOTNET_NOLOGO: true

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-dotnet@v1
        with:
          dotnet-version: "3.1.x" # SDK Version to use; x will use the latest version of the 3.1 channel
          source-url: https://nuget.pkg.github.com/bennettjones/index.json
        env:
          NUGET_AUTH_TOKEN: ${{secrets.PULL_NPM}}

      - name: Setup NuGet
        uses: NuGet/setup-nuget@v1.0.2

      - name: Cache NUGET packages
        uses: actions/cache@v2
        with:
          path: ~/.nuget/packages
          key: ${{ runner.os }}-nuget-${{ hashFiles('**/${{env.webapp_path}}.Web.csproj')}}

      - name: Download Angular app
        uses: actions/download-artifact@v2
        with:
          name: angular-app
          path: ./${{env.webapp_path}}/ClientApp/dist

      - name: Bump Angular app version
        run: |
          npm version ${{needs.build-angular.outputs.version}}
        working-directory: ./${{env.webapp_path}}/ClientApp

      - name: Bump Dotnet Version
        run: |
          dotnet tool restore
          dotnet dotnet-version --project-file ./${{env.webapp_path}}/${{env.webapp_path}}.Web.csproj --skip-vcs ${{needs.build-angular.outputs.version}}

      - name: Commit Version change
        uses: stefanzweifel/git-auto-commit-action@v4
        with:
          commit_message: Uprev to ${{needs.build-angular.outputs.version}}

      - name: Publish App
        run: dotnet publish ./${{env.webapp_path}}/${{env.webapp_path}}.Web.csproj -c TEST -o ./publish

      - name: Create Artifact
        uses: actions/upload-artifact@v2
        with:
          path: ./publish

      - name: Zip Artifact
        id: zip-artifact
        run: |
          Compress-Archive -Path .\publish\* -DestinationPath artifact.zip

      - name: Create Release
        id: create_release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: ${{needs.build-angular.outputs.version}}
          release_name: Release ${{needs.build-angular.outputs.version}}

      - name: Upload Release Asset
        uses: actions/upload-release-asset@v1.0.2
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          upload_url: ${{ steps.create_release.outputs.upload_url }}
          asset_path: ./artifact.zip
          asset_name: release-${{needs.build-angular.outputs.version}}.zip
          asset_content_type: application/zip
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/jethas-bennettjones">@jethas-bennettjones</a>,</p>
<p>Thanks for your reply!</p>
<p>I can repoduce the same on my pull request.  This is due to action <code>stefanzweifel/git-auto-commit-action@v4</code> will commit the changes together with message to repo, however, it used <code>GITHUB_TOKEN</code> by default which will not trigger new workflow. Hence, the new commit in the pull request will not have a status, which caused the pull request cannot be merged. It’s also mentioned in the action description <a href="https://github.com/stefanzweifel/git-auto-commit-action#commits-of-this-action-do-not-trigger-new-workflow-runs" rel="nofollow noopener">here</a>.</p>
<p>As a workaround, you can use another user who has write access to checkout the repository, and add job level if expression to check the <code>github.actor</code>, after the first push event workflow completed, aviod infinite workflow for the new commit.</p>
<p>It works on my side, my pull request here:<a href="https://github.com/weide-zhou/ticket13/pull/170" rel="nofollow noopener">https://github.com/weide-zhou/ticket13/pull/170</a><br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/f621cc9c6968c994f9b28bc379bb85c8232bc33d.png" data-download-href="/uploads/short-url/z7nS5aZs3Ljx9YZG0miLaHRGQs5.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181100497-7989ea27-e656-4524-af72-625741970a7b.png" alt="image" data-base62-sha1="z7nS5aZs3Ljx9YZG0miLaHRGQs5" width="690" height="341" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/f621cc9c6968c994f9b28bc379bb85c8232bc33d_2_690x341.png, https://aws1.discourse-cdn.com/github/original/2X/f/f621cc9c6968c994f9b28bc379bb85c8232bc33d.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/f/f621cc9c6968c994f9b28bc379bb85c8232bc33d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f621cc9c6968c994f9b28bc379bb85c8232bc33d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">957×474 32.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>My workflow yaml for your reference:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/weide-zhou/ticket13/tree/b564342ad37bc48a726a1ea0ff9217c7322f7eb5/.github/workflows" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/52772761/181100498-71cbdc23-48b5-4042-b9f5-724b4f56f21f.png" class="thumbnail onebox-avatar" width="420" height="420">

<h3><a href="https://github.com/weide-zhou/ticket13/tree/b564342ad37bc48a726a1ea0ff9217c7322f7eb5/.github/workflows" target="_blank" rel="nofollow noopener">weide-zhou/ticket13</a></h3>

<p>Contribute to weide-zhou/ticket13 development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Thanks</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jethas-bennettjones">@jethas-bennettjones</a>,</p>
<p>Does it work for you? Let me know if you have any queries, thanks.</p>

**Comment by jethas-bennettjones** [comment]
<p>Hi <span class="mention">@wiede-zhou</span>,</p>
<p>Thanks for following up. After reviewing the full workflow in more depth, it seems like this will actually cause more problems then it’ll solve.</p>
<p>Thank you for the time you put into this, I sincerely appreciate it.</p>
